docs/docs/integrations/providers/cratedb.mdx

```diff
@@ -0,0 +1,132 @@
+# CrateDB
+
+> [CrateDB] is a distributed and scalable SQL database for storing and
+> analyzing massive amounts of data in near real-time, even with complex
+> queries. It is PostgreSQL-compatible, based on Lucene, and inheriting
+> from Elasticsearch.
+
+
+## Installation and Setup
+
+### Setup CrateDB
+There are two ways to get started with CrateDB quickly. Alternatively,
+choose other [CrateDB installation options].
+
+#### Start CrateDB on your local machine
+Example: Run a single-node CrateDB instance with security disabled,
+using Docker or Podman. This is not recommended for production use.
+
+```bash
+docker run --name=cratedb --rm \
+  --publish=4200:4200 --publish=5432:5432 --env=CRATE_HEAP_SIZE=2g \
+  crate:latest -Cdiscovery.type=single-node
+```
+
+#### Deploy cluster on CrateDB Cloud
+[CrateDB Cloud] is a managed CrateDB service. Sign up for a
+[free trial][CrateDB Cloud Console].
+
+### Install Client
+Install the most recent version of the `langchain-cratedb` package
+and a few others that are needed for this tutorial.
+```bash
+pip install --upgrade langchain-cratedb langchain-openai unstructured
+```
+
+
+## Documentation
+For a more detailed walkthrough of the CrateDB wrapper, see
+[using LangChain with CrateDB]. See also [all features of CrateDB]
+to learn about other functionality provided by CrateDB.
+
+
+## Features
+The CrateDB adapter for LangChain provides APIs to use CrateDB as vector store,
+document loader, and storage for chat messages.
+
+### Vector Store
+Use the CrateDB vector store functionality around `FLOAT_VECTOR` and `KNN_MATCH`
+for similarity search and other purposes. See also [CrateDBVectorStore Tutorial].
+
+Make sure you've configured a valid OpenAI API key.
+```bash
+export OPENAI_API_KEY=sk-XJZ...
+```
+```python
+from langchain_community.document_loaders import UnstructuredURLLoader
```

Comment:
I believe [langchain-unstructured](https://github.com/langchain-ai/langchain-unstructured) can [load web pages](https://python.langchain.com/docs/how_to/document_loader_web/#advanced-parsing) as well.

Reply:
Thanks for this suggestion, we are always aiming to follow most recent developments in LangChain, and also recognized langchain-unstructured to be a real powerhouse.

I just tried to bring in `langchain-unstructured`, but a drawback to use it for minimal examples could be the large array of dependencies it pulls in.

```
$ uv pip install langchain-unstructured
Resolved 45 packages in 598ms
Prepared 5 packages in 3.44s
Uninstalled 2 packages in 19ms
Installed 16 packages in 86ms
 + aiofiles==24.1.0
 + coloredlogs==15.0.1
 + cryptography==44.0.0
 + eval-type-backport==0.2.2
 + flatbuffers==24.12.23
 + humanfriendly==10.0
 + jsonpath-python==1.0.6
 + langchain-unstructured==0.1.6
 + mpmath==1.3.0
 + onnxruntime==1.19.2
 + protobuf==5.29.2
 - pydantic==2.10.4
 + pydantic==2.9.2
 - pydantic-core==2.27.2
 + pydantic-core==2.23.4
 + pypdf==5.1.0
 + sympy==1.13.3
 + unstructured-client==0.28.1
```